### PR TITLE
--test-command was broken

### PR DESCRIPTION
**Pull request by chick**
fix --test-command so that it is used in verilator and vcs backends
add test to DriverSpec to illustrate it's use.
added two other tests to DriverSpec to illustrate overriding directories

### CODE REVIEW

File: src/main/scala/chisel3/iotesters/VCSBackend.scala [hunk 1]
```diff
@@ -143,7 +143,13 @@ private[iotesters] object setupVCSBackend {
     genVCSVerilogHarness(dut, new FileWriter(vcsHarnessFile), vpdFile.toString)
     assert(verilogToVCS(circuit.name, dir, new File(vcsHarnessFileName)).! == 0)
 
-    (dut, new VCSBackend(dut, Seq((new File(dir, circuit.name)).toString)))
+    val command = if(optionsManager.testerOptions.testCmd.nonEmpty) {
+      optionsManager.testerOptions.testCmd
+    } else {
+      Seq((new File(dir, s"V${circuit.name}")).toString)
```
Comment (ucbjrl) [inline src/main/scala/chisel3/iotesters/VCSBackend.scala]: This looks like a copy/paste typo:

    Seq((new File(dir, s"V${circuit.name}")).toString)

should be

    Seq((new File(dir, circuit.name)).toString)

no?

Comment (chick) [inline src/main/scala/chisel3/iotesters/VCSBackend.scala]: Sure seems like you are correct, and yet the test passed.  I'll look into it

Comment (chick) [inline src/main/scala/chisel3/iotesters/VCSBackend.scala]: @ucbjrl oh, yeah, vcs backend is not actually tested as we discussed this morning.  We need the mechanism to see if vcs supported in platform running the tests

Comment (chick) [inline src/main/scala/chisel3/iotesters/VCSBackend.scala]: @ucbjrl great catch btw

**Pull request merged**